addons/l10n_es_edi_tbai/models/ir_attachment.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from odoo.addons.l10n_es_edi_tbai.models.web_services import get_web_services_for_agency
+from odoo.tools import xml_utils
+
+
+class IrAttachment(models.Model):
+    _inherit = 'ir.attachment'
+
+    def _l10n_es_tbai_load_xsd_attachments(self, agency=None):
+        """
+        Downloads the xsd validation files if they don't already exist.
+        The default tax agency is provided by self.env.company.
+        """
+        xml_utils.load_xsd_files_from_url(
+            self.env, 'https://www.w3.org/TR/xmldsig-core/xmldsig-core-schema.xsd', 'xmldsig-core-schema.xsd',
+            xsd_name_prefix='l10n_es_edi_tbai')
+
+        if agency is None:
+            agency = self.env.company.l10n_es_tbai_tax_agency
+
+        web_services = get_web_services_for_agency(agency=agency, env=self.env)
+        urls = web_services.get_xsd_urls()
+        names = web_services.get_xsd_names()
+        # For Bizkaia, one url per file (same order for urls and names)
+        if isinstance(urls, tuple):
+            for url, name in zip(urls, names):
+                xml_utils.load_xsd_files_from_url(
+                    self.env, url, name,
+                    xsd_name_prefix='l10n_es_edi_tbai',
+                )
+        # For other agencies, single url to zip file (only keep the desired names)
+        else:
+            xml_utils.load_xsd_files_from_url(
+                self.env, urls, agency + '_XSDs.zip',
+                xsd_name_prefix='l10n_es_edi_tbai',
+                xsd_names_filter=list(names.values()),
```

Comment:
```suggestion
```
@odoostan

Reply:
@jco-odoo while this does ensure _all_ XSDs are saved from the zip (Araba and Bizkaia have zips for XSDs), it does not resolve the issue.
```

```odoo.tools.xml_utils: Retrieved attachment from database, with name: xmldsig-core-schema.xsd 
odoo.tools.xml_utils: Retrieved attachment from database, with name: araba_XSDs.zip 
odoo.tools.xml_utils: The XSD file(s) could not be found, even after a reload 
```

Standalone XSDs are found
Original zips are found (not even sure why we keep them but OK)
XSDs extracted from the zip are not found